src/neo/SmartContract/ApplicationEngine.Runtime.cs

```diff
@@ -293,8 +293,10 @@ protected internal BigInteger GetRandom()
         protected internal void RuntimeLog(byte[] state)
         {
             if (state.Length > MaxNotificationSize) throw new ArgumentException(null, nameof(state));
-            string message = Utility.StrictUTF8.GetString(state);
-            Log?.Invoke(this, new LogEventArgs(ScriptContainer, CurrentScriptHash, message));
+            var log = new LogEventArgs(ScriptContainer, CurrentScriptHash, Utility.StrictUTF8.GetString(state));
+            Log?.Invoke(this, log);
+            _logs ??= new List<LogEventArgs>();
+            _logs.Add(log);
```

Comment:
I think logs can be collected by plugins. It shouldn't be done by the core.

Reply:
will work on that later